tutorials/intermediate/autoscale.py

```diff
@@ -0,0 +1,178 @@
+"""
+Autoscaling
+===========
+
+The limits on an axis can be set manually (e.g. ``ax.set_xlim(xmin, xmax)``)
+or Matplotlib can set them automatically based on the data already on the axes.
+There are a number of options to this autoscaling behaviour, discussed below.
+"""
+
+###############################################################################
+# We will start with a simple line plot showing that autoscaling
+# extends the axis limits 5% beyond the data limits (-2π, 2π).
+
+import numpy as np
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+
+x = np.linspace(-2 * np.pi, 2 * np.pi, 100)
+y = np.sinc(x)
+
+fig, ax = plt.subplots()
+ax.plot(x, y)
+fig.show()
+
+###############################################################################
+# Margins
+# -------
+# The default margin around the data limits is 5%:
+
+ax.margins()
+
+###############################################################################
+# The margins can be made larger using `~matplotlib.axes.Axes.margins`:
+
+fig, ax = plt.subplots()
+ax.plot(x, y)
+ax.margins(0.2, 0.2)
+fig.show()
+
+###############################################################################
+# In general, margins can be in the range (-0.5, ∞), where negative margins set
+# the axes limits to a fraction of the data range, and allow a zoom effect.
+# Using a single number for margins affects both axes, a single margin can be
+# customized using keyword arguments ``x`` or ``y``, but positional and keyword
+# interface cannot be combined.
+
+fig, ax = plt.subplots()
+ax.plot(x, y)
+ax.margins(y=-0.2)
+fig.show()
+
+###############################################################################
+# Sticky edges
+# ------------
+# Some :class:`.Artist` objects do not allow margins.  For example, setting
+# ``margin=0.2`` on ``plt.imshow`` does not affect the resulting plot.
+#
+
+xx, yy = np.meshgrid(x, x)
+zz = np.sinc(np.sqrt((xx - 1)**2 + (yy - 1)**2))
+
+fig, ax = plt.subplots(ncols=2, figsize=(12, 8))
+ax[0].imshow(zz)
+ax[0].set_title("margins unchanged")
```

Comment:
```suggestion
ax[0].set_title("default margins")
```

Also all following uses of "unchanged" below.

Reply:
"unchanged" removed.